### PR TITLE
feat: return estimated balance in account balance endpoints

### LINKED ISSUE #2035: Add property for estimated balance to balances endpoint
As discussed [here](https://github.com/hirosystems/stacks-blockchain-api/issues/2026), Leather would benefit from an `estimated_balance` property in addition to the `balance` as returned by the `balances` endpoint for a given Stacks address.

This estimated balance would subtract the total of all outbound transfers from its confirmed balance for a given address.

Leather currently calculates such a balance through two separate queries and a final calculation:

1. fetch balance from `api.hiro.so/extended/v1/address/<address>/balances`
2. fetch pending txs from `api.hiro.so/extended/v1/tx/mempool?address=<address>&limit=50`
3. do calculation: `availableBalance = balance - sum of all outbound transaction amounts`

### PR DESCRIPTION
**Pull request by rafaelcr**
This PR adds an optional `estimated_balance` entry to the balance endpoint responses which reflects the current confirmed STX balance plus any STX delta (sent or received) in transactions currently visible in the mempool.

Fixes #2035

### CODE REVIEW

File: src/datastore/pg-store.ts [hunk 1]
```diff
@@ -2476,6 +2476,36 @@ export class PgStore extends BasePgStore {
...
   }
 
+  /**
+   * Returns the total STX balance delta affecting a principal from transactions currently in the
+   * mempool.
+   */
+  async getPrincipalMempoolStxBalanceDelta(sql: PgSqlClient, principal: string): Promise<bigint> {
+    const results = await sql<{ delta: string }[]>`
+      WITH sent AS (
+        SELECT token_transfer_amount + fee_rate AS total
```
Comment (zone117x) [inline src/datastore/pg-store.ts:2486]: What happens here with tx types that have `token_transfer_amount==null`, for example contract-calls?

Comment (rafaelcr) [reply src/datastore/pg-store.ts]: Good catch, done

File: src/datastore/pg-store.ts [hunk 2]
```diff
@@ -2483,14 +2483,14 @@ export class PgStore extends BasePgStore {
   async getPrincipalMempoolStxBalanceDelta(sql: PgSqlClient, principal: string): Promise<bigint> {
     const results = await sql<{ delta: string }[]>`
       WITH sent AS (
-        SELECT token_transfer_amount + fee_rate AS total
+        SELECT token_transfer_amount + COALESCE(fee_rate, 0) AS total
```
Comment (zone117x) [inline src/datastore/pg-store.ts:2486]: This can end up running `null + <number>` right, does it need `COALESCE(token_transfer_amount, 0)`?

Comment (zone117x) [reply src/datastore/pg-store.ts:2486]: Might be useful to add another type of tx in the tests

Comment (rafaelcr) [reply src/datastore/pg-store.ts:2486]: Oh duh, added the coalesce to the wrong field lol... yep I will add a test for this

**Review by zone117x — APPROVED** [review]
Great job with this feature!

**Pull request merged**